exasol_transformers_extension/deploy.py

```diff
@@ -1,29 +1,90 @@
-    scripts_deployer_main
-from exasol.python_extension_common.deployment.language_container_deployer_cli \
-    import language_container_deployer_main, slc_parameter_formatters, CustomizableParameters
-from exasol_transformers_extension.deployment.te_language_container_deployer import TeLanguageContainerDeployer
+from exasol.python_extension_common.cli.std_options import (
+    StdParams, StdTags, select_std_options, ParameterFormatters, make_option_secret)
+from exasol.python_extension_common.cli.language_container_deployer_cli import (
+    LanguageContainerDeployerCli)
+from exasol.python_extension_common.cli.bucketfs_conn_object_cli import BucketfsConnObjectCli
+from exasol.python_extension_common.connections.pyexasol_connection import open_pyexasol_connection
+from exasol.python_extension_common.connections.bucketfs_location import (
+    ConnectionInfo, write_bucketfs_conn_object)
+from exasol_transformers_extension.deployment.te_language_container_deployer import (
+    TeLanguageContainerDeployer)
+from exasol_transformers_extension.deployment.scripts_deployer import ScriptsDeployer
 
+DEPLOY_SLC_ARG = 'deploy_slc'
+DEPLOY_SCRIPTS_ARG = 'deploy_scripts'
+CONTAINER_URL_ARG = 'container_url'
+CONTAINER_NAME_ARG = 'container_name'
+BUCKETFS_CONN_NAME_ARG = 'bucketfs_conn_name'
+TOKEN_CONN_NAME_ARG = 'token_conn_name'
+TOKEN_ARG = 'token'
 
-@click.group()
-def main():
-    pass
+ver_formatter = ParameterFormatters()
+ver_formatter.set_formatter(CONTAINER_URL_ARG, TeLanguageContainerDeployer.SLC_URL_FORMATTER)
+ver_formatter.set_formatter(CONTAINER_NAME_ARG, TeLanguageContainerDeployer.SLC_NAME)
+formatters = {StdParams.version: ver_formatter}
 
 
-slc_parameter_formatters.set_formatter(CustomizableParameters.container_url,
-                                       TeLanguageContainerDeployer.SLC_URL_FORMATTER)
-slc_parameter_formatters.set_formatter(CustomizableParameters.container_name,
-                                       TeLanguageContainerDeployer.SLC_NAME)
+def get_opt_name(arg_name: str) -> str:
+    # This and the next function should have been implemented in the PEC.
+    return f'--{arg_name.replace("_", "-")}'
 
-main.add_command(scripts_deployer_main)
-main.add_command(language_container_deployer_main)
+
+def get_bool_opt_name(arg_name: str) -> str:
+    opt_name = arg_name.replace("_", "-")
+    return f'--{opt_name}/--no-{opt_name}'
+
+
+opt_lang_alias = {'type': str, 'default': 'PYTHON3_TE'}
+opt_token = {'type': str, 'help': 'Huggingface token for private models'}
+make_option_secret(opt_token, prompt='Huggingface token')
+opts = select_std_options([StdTags.DB, StdTags.BFS, StdTags.SLC],
+                          formatters=formatters, override={StdParams.language_alias: opt_lang_alias})
+opts.append(click.Option([get_bool_opt_name(DEPLOY_SLC_ARG)], type=bool, default=True,
+                         help='Deploy SLC'))
+opts.append(click.Option([get_bool_opt_name(DEPLOY_SCRIPTS_ARG)], type=bool, default=True,
```

Comment:
i am curious. why do you do the arguments like this instead of a standard click option?

Reply:
The answer depends on what you mean by a standard click option.
We could have added these two to the `StdParams` in PEC, if you are referring to this kind of standard options. At least the first one - `deploy_slc`. If there is an SLC the must be a way to skip its deployment. The other one - `deploy_scripts` - I am not sure. It doesn't fit into any group. We can have an extension without any scripts, in theory.